benchmark/make.jl

```diff
@@ -1,66 +1,142 @@
+            opt = OPTIMIZERS[name]
+            suite[name][i, j] = @benchmarkable einexpr($opt, $expr)
+            count[name][i, j] = mapreduce(flops, +, Branches(einexpr(opt, expr)))
+        end
+    end
+
+    # tune benchmarks
+    tune!(suite; verbose=true)
 
-suite["naive"][1] = @benchmarkable einexpr(EinExprs.Naive(), $expr)
-suite["exhaustive"][1] = @benchmarkable einexpr(Exhaustive(), $expr)
-suite["greedy"][1] = @benchmarkable einexpr(Greedy(), $expr)
-suite["kahypar"][1] = @benchmarkable einexpr(HyPar(), $expr)
-
-# BENCHMARK 2
-A = EinExpr([:A, :a, :b, :c], Dict(:A => 2, :a => 2, :b => 2, :c => 2))
-B = EinExpr([:b, :d, :e, :f], Dict(:b => 2, :d => 2, :e => 2, :f => 2))
-C = EinExpr([:a, :e, :g, :C], Dict(:a => 2, :e => 2, :g => 2, :C => 2))
-D = EinExpr([:c, :h, :d, :i], Dict(:c => 2, :h => 2, :d => 2, :i => 2))
-E = EinExpr([:f, :i, :g, :j], Dict(:f => 2, :i => 2, :g => 2, :j => 2))
-F = EinExpr([:B, :h, :k, :l], Dict(:B => 2, :h => 2, :k => 2, :l => 2))
-G = EinExpr([:j, :k, :l, :D], Dict(:j => 2, :k => 2, :l => 2, :D => 2))
-expr = sum([A, B, C, D, E, F, G], skip = [:A, :B, :C, :D])
-
-suite["naive"][2] = @benchmarkable einexpr(EinExprs.Naive(), $expr)
-suite["exhaustive"][2] = @benchmarkable einexpr(Exhaustive(), $expr)
-suite["greedy"][2] = @benchmarkable einexpr(Greedy(), $expr)
-suite["kahypar"][2] = @benchmarkable einexpr(HyPar(), $expr)
-
-# Tuning
-tune!(suite)
-
-# Run
-GC.enable(false)
-results = run(suite, verbose = true)
-GC.enable(true)
-
-using BenchmarkPlots, StatsPlots
-
-for (method, group) in results
-    plt = plot(
-        results[method],
-        yaxis = (:log10, "Execution time [ns]"),
-        xaxis = (:flip, "Benchmark set"),
-        title = "$method",
+    # run benchmarks
+    results = run(suite, verbose = true)
+
+    # construct plots
+    x = Vector{Float64}[]
+    y = Vector{Float64}[]
+
+    for i in 1:k
+        push!(x, Float64[])
+        push!(y, Float64[])
+    end
+        
+    for name in optimizers, i in 1:k
+        xx = 0.0
+        yy = 0.0
+
+        for j in 1:m
+            xx += count[name][i, j] / m
+            yy += time(minimum(results[name][i, j])) / m
+        end
+
+        push!(x[i], xx)
+        push!(y[i], yy)
+    end
+
+    figure = Figure(; size=(600, 200 * k))
+
+    for i in 1:k - 1
+        axis = Axis(figure[i, 1];
+            ylabel = "time (ns)",
+            xscale=log10,
+            yscale=log10,
+            xautolimitmargin = (0.1, 0.2),
+            yautolimitmargin = (0.1, 0.2),
+        )
+
+        scatter!(axis, x[i], y[i])
+        text!(axis, x[i], y[i]; text=optimizers)
+    end
+
+    axis = Axis(figure[k, 1];
+        ylabel = "time (ns)",
+        xlabel = "flops",
+        xscale=log10,
+        yscale=log10,
+        xautolimitmargin = (0.1, 0.2),
+        yautolimitmargin = (0.1, 0.2),
     )
-    display(plt)
+
+    scatter!(axis, x[k], y[k])
+    text!(axis, x[k], y[k]; text=optimizers)
+
+    save("$n.png", figure)
```

Comment:
do you mind saving them as SVG? PNG files are binary and these plots are small enough to save them as text formatted as SVG, so Git won't have problems with them.

```suggestion
    save("$n.svg", figure)
```

Reply:
done!